### PR TITLE
dasm: init at 2.20.13

### PR DESCRIPTION
**Pull request by jjwatt**
###### Motivation for this change
Add the "dasm" assembler to nixpkgs.

###### Things done
- Added dasm package
- Added dasm to top-level/all-packages.nix
- Added myself, "jjwatt" to the maintainers-list

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/dasm/default.nix
Comment (cole-h) [inline pkgs/development/compilers/dasm/default.nix:29]:
```suggestion
		homepage = "https://dasm-assembler.github.io";
```

For RFC45 compliance.

File: pkgs/development/compilers/dasm/default.nix
Comment (cole-h) [inline pkgs/development/compilers/dasm/default.nix:10]:
```suggestion
		rev = version;
```

We prefer to pin to the tagged release version.

File: pkgs/development/compilers/dasm/default.nix
Comment (cole-h) [inline pkgs/development/compilers/dasm/default.nix:26]:
```suggestion
```

**Review by cole-h — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/dasm/default.nix
Comment (jjwatt) [reply pkgs/development/compilers/dasm/default.nix:10]: Ah of course. Forgot I could refer to tags on github like that, Makes sense cus they're git refs. Thanks.

File: pkgs/development/compilers/dasm/default.nix
Comment (cole-h) [inline pkgs/development/compilers/dasm/default.nix:10]:
```suggestion
		rev = version;
```

I meant literally put the variable here. This way you don't need to change two variables when updating this `dasm`, only the first one.

Comment (jjwatt) [reply pkgs/development/compilers/dasm/default.nix]: Oh! Beautiful. Thanks, again!

File: maintainers/maintainer-list.nix
Comment (cole-h) [inline maintainers/maintainer-list.nix:3771]:
```suggestion
    githubId = 2397327;
```

Just to fall in line with the rest of the file ;)

**Review by cole-h — CHANGES_REQUESTED** [review]
Final couple requested changes:

It would be nice if you could 1) squash your commits back into the original `init at` commit, and 2) separate your maintainer changes into a separate commit: `maintainers: add jwatt`.

Thanks for bearing with my nitpicks.

**Comment by jjwatt** [comment]
Didn't know that I could do that in a review/pull-request! I appreciate your time and feedback! I don't consider them nits. I haven't worked this deeply with github before and am used to gerrit, plus I'm new to nix committing, so this is a learning experience.

er, I assume by splitting and squashing you mean with a good ol' git rebase. I'll try that and push it and see if it works.

**Comment by jjwatt** [comment]
That seemed to work, but I somehow lost the use of version var change that you suggested. Lemme add that back in.

File: pkgs/development/compilers/dasm/default.nix
Comment (cole-h) [inline pkgs/development/compilers/dasm/default.nix:1]: You don't consider my suggestions nits? Well then, here's a nit for ya:

```suggestion
{ stdenv, fetchFromGitHub }:
```

(please :) )

Comment (jjwatt) [reply pkgs/development/compilers/dasm/default.nix]: lol. Nope! bring it on :)

ahh. I just discovered `nixfmt`, too.

**Review by cole-h — APPROVED** [review]
LGT my untrained eye! Even ran a little test program and it compiles/runs fine. Should be good to go, assuming nobody has more nits! Though it might take a bit to get merged; be patient, ~~bug somebody on IRC,~~ or post in the Discourse thread for reviewed PRs: https://discourse.nixos.org/t/prs-already-reviewed/2617

```
[3 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/84739
32 package marked as broken and skipped:
clang-sierraHack digitalbitbox gdata-sharp iterm2 linuxPackages_4_4.evdi linuxPackages_hardkernel_4_
14.bcc linuxPackages_hardkernel_4_14.bpftrace linuxPackages_hardkernel_4_14.can-isotp linuxPackages_
hardkernel_4_14.chipsec linuxPackages_hardkernel_4_14.digimend linuxPackages_hardkernel_4_14.evdi li
nuxPackages_hardkernel_4_14.mba6x_bl linuxPackages_hardkernel_4_14.nvidia_x11 linuxPackages_hardkern
el_4_14.nvidia_x11_legacy390 linuxPackages_hardkernel_4_14.prl-tools nix-linter octave-jit octoprint
 php73Extensions.zmq php74Extensions.couchbase php74Extensions.pcs php74Extensions.pthreads php74Ext
ensions.zmq python27Packages.caffe python27Packages.habanero python27Packages.handout python27Packag
es.hass-nabucasa python37Packages.nixpart python38Packages.libselinux python38Packages.nixpart pytho
n38Packages.poezio python38Packages.pyblock

1 package built:
dasm
```

---

Test program:

```asm
str set "Hello Nix!"
 echo str
```

**Review by JohnAZoidberg — APPROVED** [review]
> Test program:
>
> ```str set "Hello Nix!"
>  echo str
> ```

Can confirm that it works!

File: pkgs/development/compilers/dasm/default.nix
Comment (JohnAZoidberg) [inline pkgs/development/compilers/dasm/default.nix:24]:
```suggestion
	checkTarget = "test";
```

File: pkgs/development/compilers/dasm/default.nix
Comment (JohnAZoidberg) [inline pkgs/development/compilers/dasm/default.nix:21]:
```suggestion
		patchShebangs ./test/
```

**Review by JohnAZoidberg — CHANGES_REQUESTED** [review]
The test directory and make target are called `test`, not `tests`. If you also add `doCheck`, nix will run the tests :)

Please also use 2 spaces for indentation instead of tabs.

File: pkgs/development/compilers/dasm/default.nix
Comment (JohnAZoidberg) [reply pkgs/development/compilers/dasm/default.nix]: Yeah, nixfmt is cool but it's not official yet. I.e. it's not the standard for nixpkgs.

**Comment by JohnAZoidberg** [comment]
image
If you tick this box, we maintainers would be able to fix those small nitpicks for you (if we feel like it).

**Comment by jjwatt** [comment]
It looks like it is checked on my end.

**Comment by JohnAZoidberg** [comment]
I squashed the commits and pushed them as fbb76f6ace21d8f1a3956e6f329e404cad4f6136.

Thanks for the contribution!

**Pull request closed without merging**
